### ISSUE TITLE
[YSQL] Live connection aren't dropped when database is dropped

### ISSUE THREAD

**Issue by ddorian** [yugabyte/yugabyte-db#7119]
On 2.5.1.0, if I drop a database which has connections to it, the database is dropped but the connections stay orphan.

On PostgreSQL, you can't drop a database with active connections, so this is not possible. 

Simple script in python that creates/drops a database in a loop to show connections remaining open:

```python
def schema_test():
    import psycopg2
    index = 0
    pg0 = psycopg2.connect("dbname=yugabyte host=localhost user=yugabyte password=yugabyte port=5433")
    pg0.autocommit = True

    c0 = pg0.cursor()
    while True:
        c0.execute("create database vid_app;")
        pg1 = psycopg2.connect("dbname=vid_app host=localhost user=yugabyte password=yugabyte port=5433")
        pg1.autocommit = True
        c1 = pg1.cursor()
        c1.execute("create table aa(id text primary key);")
        c0.execute("drop database vid_app;")
        pg1.close()
        index += 1
        print(f"index:{index}")
    exit()

schema_test()

```

You can see processes increasing by running:

```
ps ax |grep yugabyte
```

**Comment by kmuthukk** [comment]
@ddorian -- would you mind verifying this with 2.5.2?

**Comment by ddorian** [comment]
@kmuthukk it's fixed on 2.5.2.0. Connections get closed correctly.

**Comment by kmuthukk** [comment]
Related to/duplicate of: https://github.com/yugabyte/yugabyte-db/issues/7105
